types/networks/src/lib.rs

```diff
@@ -302,3 +304,68 @@ impl Default for ChainConfig {
         }
     }
 }
+
+pub fn default_network_version() -> NetworkVersion {
+    NetworkVersion::V15
+}
+
+pub mod de_network_version {
+    use fil_types::NetworkVersion;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<NetworkVersion, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let version: &str = Deserialize::deserialize(deserializer)?;
+        let version = version.to_lowercase();
+
+        match version.as_str() {
+            "v0" => Ok(NetworkVersion::V0),
+            "v1" => Ok(NetworkVersion::V1),
+            "v2" => Ok(NetworkVersion::V2),
+            "v3" => Ok(NetworkVersion::V3),
+            "v4" => Ok(NetworkVersion::V4),
+            "v5" => Ok(NetworkVersion::V5),
+            "v6" => Ok(NetworkVersion::V6),
+            "v7" => Ok(NetworkVersion::V7),
+            "v8" => Ok(NetworkVersion::V8),
+            "v9" => Ok(NetworkVersion::V9),
+            "v10" => Ok(NetworkVersion::V10),
+            "v11" => Ok(NetworkVersion::V11),
+            "v12" => Ok(NetworkVersion::V12),
+            "v13" => Ok(NetworkVersion::V13),
+            "v14" => Ok(NetworkVersion::V14),
+            "v15" => Ok(NetworkVersion::V15),
+            _ => Ok(NetworkVersion::V15),
```

Comment:
Better produce an error here. We don't want to allow user to use an unexisting version or have erroneous version.
ie. `"nv16"` or `"V16"` should lead to a deserialization error and not return `NetworkVersion::V15`